bundles/org.openhab.binding.draytonwiser/src/main/java/org/openhab/binding/draytonwiser/internal/api/DraytonWiserApiException.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.draytonwiser.internal.api;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Exception thrown in case of api problems.
+ *
+ * @author Hilbrand Bouwkamp - Initial contribution
+ */
+@NonNullByDefault
+public class DraytonWiserApiException extends RuntimeException {
```

Comment:
Why isn't this a checked exception?

Reply:
Because it isn't something that can be recovered from in general, as it is thrown when the heathub is refusing to respond for whatever reason.

Reply:
@Hilbrand Do you have an opinion on this?

Reply:
No particular reason. I guess because it's a low level method it means it a lot of methods have to throw the exception if checked, and that sometimes makes code more difficult (to read) when used with streams. But that doesn't seem to be a problem here. So I changed it to a checked exception in the pull request I made.

Reply:
Resolved in cc3fafd